### PR TITLE
Create a callback to visualise the segmentation results

### LINKED ISSUE #385: Create a callback to visualise the segmentation results
We would like to visualise the predictions of the segmentation model in order to track its progress qualitatively.

### PR DESCRIPTION
**Pull request by ioangatop**
Closes #385

ezgif-7-23f4126422-1

### CODE REVIEW

**Review by nkaenzig — COMMENTED** [review]
This looks great! I left a couple of minor comments.

File: src/eva/vision/callbacks/loggers/batch/segmentation.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        image_grid = _make_grid_from_outputs(images, targets, predictions)
+
+        log.log_image(
+            trainer.loggers,
+            image=image_grid,
+            tag=tag,
+            step=trainer.global_step,
+        )
+
+    def _denomalize_image(self, image: torch.Tensor) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (ioangatop) [reply src/eva/vision/callbacks/loggers/batch/segmentation.py:83]: nice done

File: src/eva/vision/callbacks/loggers/batch/segmentation.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        mean: Tuple[float, ...] = (0.5, 0.5, 0.5),
+        std: Tuple[float, ...] = (0.5, 0.5, 0.5),
+        log_every_n_epochs: int | None = 10,
+        log_every_n_steps: int | None = None,
+    ) -> None:
+        """Initializes the callback object.
+
+        Args:
+            max_samples: The maximum number of images displayed in the grid.
+            number_of_rows: Number of images displayed in each row of the grid.
```
[the remark this answers is not in the document]
Comment (ioangatop) [reply src/eva/vision/callbacks/loggers/batch/segmentation.py:35]: good catch, its also was not used! changed to `number_of_subgrid_rows`, meaning number of rows for each item subgrid (images, targets, predictions)

File: src/eva/vision/callbacks/loggers/batch/segmentation.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    """Draws a semantic mask to an image RGB tensor.
+
+    The input semantic mask is a (H x W) shaped tensor with
+    integer values which represent the pixel class id.
+
+    Args:
+        tensor: An image tensor of range [0., 1.].
+
+    Returns:
+        The image as a numpy tensor in the range of [0., 255.].
```
[the remark this answers is not in the document]
Comment (ioangatop) [reply src/eva/vision/callbacks/loggers/batch/segmentation.py:145]: nice done!

**Review by nkaenzig — APPROVED** [review]

File: src/eva/vision/callbacks/loggers/batch/segmentation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ioangatop) [reply src/eva/vision/callbacks/loggers/batch/segmentation.py:35]: `number_of_images_per_subgrid_row`

File: src/eva/vision/callbacks/loggers/batch/base.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        self._log_every_n_epochs = log_every_n_epochs
+        self._log_every_n_steps = log_every_n_steps
+
+    @override
+    def on_sanity_check_start(
+        self,
+        trainer: pl.Trainer,
+        pl_module: pl.LightningModule,
+    ) -> None:
+        self._force_log = False
```
[the remark this answers is not in the document]
Comment (ioangatop) [reply src/eva/vision/callbacks/loggers/batch/base.py:51]: dev leftovers - removed!

**Pull request merged**